bin/tx-prover/src/proxy/mod.rs

```diff
@@ -300,22 +305,27 @@ static QUEUE: Lazy<RequestQueue> = Lazy::new(RequestQueue::new);
 /// Custom context for the request/response lifecycle
 /// We use this context to keep track of the number of tries for a request, the unique ID for the
 /// request, and the worker that will process the request.
+#[derive(Debug)]
 pub struct RequestContext {
     /// Number of tries for the request
     tries: usize,
     /// Unique ID for the request
     request_id: Uuid,
     /// Worker that will process the request
     worker: Option<Worker>,
+    /// Parent span for the request
+    parent_span: Span,
 }
 
 impl RequestContext {
     /// Create a new request context
     fn new() -> Self {
+        let request_id = Uuid::new_v4();
         Self {
             tries: 0,
-            request_id: Uuid::new_v4(),
+            request_id,
             worker: None,
+            parent_span: info_span!("proxy:new_request", request_id = request_id.to_string()),
```

Comment:
Question: why don't we need to specify `target` both here and in other places in this file?

Reply:
We **should** be using `target` here. For the "sons" (the traces that we create in each method of `HttpProxy`) it is not needed since there it gets inherited.